plugins/module_utils/ilo_redfish_utils.py

```diff
@@ -51,6 +99,403 @@ def get_ilo_sessions(self):
+            }
+
+        return result
+
+    def get_smartstorage_physical_drives(self):
+        # This method returns list of physical drives present in the OOB controller
+        physical_drives = {}
+        physical_drives_count = 0
+        result = {}
+
+        response = self.get_request(self.root_uri + self.systems_uri)
+        if not response["ret"]:
+            return response
+
+        response = self.get_request(self.root_uri + self.systems_uri + "SmartStorage/ArrayControllers/")
+
+        if not response["ret"]:
+            return response
+
+        data = response["data"]
+        if data["Members@odata.count"] == 0:
+            # return physical_drives, physical_drives_count
+            result["physical_drives"] = {}
+            result["physical_drives_count"] = 0
+            return {
+                "ret": True,
+                "msg": result
+            }
+
+        # Get Members of ArrayControllers
+        for mem in data["Members"]:
+            physical_drive_list = []
+            array_url = mem["@odata.id"]
+            response = self.get_request(
+                self.root_uri + mem["@odata.id"]
+            )
+
+            if not response["ret"]:
+                return response
+
+            data = response["data"]
+            if 'Links' in data and 'PhysicalDrives' in data['Links']:
+                log_url = data['Links']['PhysicalDrives']['@odata.id']
+            elif 'links' in data and 'PhysicalDrives' in data['links']:
+                log_url = data['links']['PhysicalDrives']['href']
+            else:
+                return {
+                    "ret": False,
+                    "msg": "Physical drive URI not found in %s response: %s" % (mem["@odata.id"], data)
+                }
+
+            # Get list of physical drives URI
+            response = self.get_request(
+                self.root_uri + log_url
+            )
+
+            if not response["ret"]:
+                return response
+
+            json_data = response["data"]
+            for entry in json_data["Members"]:
+                # Get each physical drives details
+                response = self.get_request(
+                    self.root_uri + entry["@odata.id"]
+                )
+
+                if not response["ret"]:
+                    return response
+
+                log_data = self.remove_odata(response["data"])
+                physical_drive_list.append(log_data)
+            physical_drives.update({"array_controller_" + str(array_url.split("/")[-2]): physical_drive_list})
+            physical_drives_count = physical_drives_count + len(physical_drive_list)
+            result["physical_drives"] = physical_drives
+            result["physical_drives_count"] = physical_drives_count
+
+        return {
+            "msg": result,
+            "ret": True
+        }
+        # return result
+
+    def get_logical_drives_details(self, array_controllers=False):
```

Comment:
Can we make the same change here (and other places where "SmartStorage" is the target Storage subsystem) to include "smartstorage" in the method name?

Reply:
Updated "get_logical_drives_details" to "get_smartstorage_logical_drives". Keeping other function names such as "create_logical_drives" the same as they don't have an equivalent with storage or simaplstorage in redfish_utils

Reply:
Resolving this conversation, thanks!